glide-core/src/socket_listener.rs

```diff
@@ -311,6 +312,44 @@ async fn send_command(
         .map_err(|err| err.into())
 }
 
+// Parse the cluster scan command parameters from protobuf and send the command to redis-rs.
+async fn cluster_scan(cluster_scan: ClusterScan, mut client: Client) -> ClientUsageResult<Value> {
+    // Since we don't send the cluster scan as a usual command, but through a special function in redis-rs library,
+    // we need to handle the command separately.
+    // Specifically, we need to handle the cursor, which is not the cursor returned from the server,
+    // but the ID of the ScanStateRC, stored in the cluster scan container.
+    // We need to get the ref from the table or create a new one if the cursor is empty.
+    let cursor: String = cluster_scan.cursor.into();
+    let cluster_scan_cursor = if cursor.is_empty() {
+        ScanStateRC::new()
+    } else {
+        get_cluster_scan_cursor(cursor)?
+    };
+
+    let match_pattern_string = cluster_scan
+        .match_pattern
+        .map(|pattern| pattern.to_string());
+    let match_pattern = match_pattern_string.as_deref();
+    let count = cluster_scan.count.map(|count| count as usize);
+
+    let object_type =
+        cluster_scan
+            .object_type
+            .and_then(|object_type| match object_type.to_string().as_str() {
+                "String" => Some(redis::ObjectType::String),
+                "List" => Some(redis::ObjectType::List),
+                "Set" => Some(redis::ObjectType::Set),
+                "ZSet" => Some(redis::ObjectType::ZSet),
+                "Hash" => Some(redis::ObjectType::Hash),
```

Comment:
Should Stream be a type here?

Reply:
Yes, its a mistake, will be added. Need to be added at redis-rs side so create the api but dont test it yet. Ill all tomorrow.

Reply:
But it'll be fixed to return an error so tests will not pass if not right params will be provided